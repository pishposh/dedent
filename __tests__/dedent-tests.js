// @flow

import dd from '../dedent';

describe("dedent", () => {
  it("works without interpolation", () => {
    expect(
      dd`first
         second
         third`
    ).toMatchSnapshot();
  });

  it("works with interpolation", () => {
    expect(
      dd`first ${"line"}
         ${"second"}
         third`
    ).toMatchSnapshot();
  });

  it("works with suppressed newlines", () => {
    expect(
      dd`first \
         ${"second"}
         third`
    ).toMatchSnapshot();
  });

  it("works with multiline interpolation", () => {
    expect(
      dd`
        first
          ${"second\nthird"}
        fourth
      `
    ).toMatchSnapshot();
  });

  it("works with inline interpolation", () => {
    expect(
      dd`
        first ${"second\nthird"} fourth
      `
    ).toMatchSnapshot();
  });

  it("works with dedent interpolation", () => {
    expect(
      dd`
        first
          ${dd`
            second
            third
          `}
        fourth
      `
    ).toMatchSnapshot();
  });

  it("works with blank first line", () => {
    expect(dd`
      Some text that I might want to indent:
        * reasons
        * fun
      That's all.
    `).toMatchSnapshot();
  });

  it("works with multiple blank first lines", () => {
    expect(
      dd`

         first
         second
         third`
    ).toMatchSnapshot();
  });

  it("works with removing same number of spaces", () => {
    expect(
      dd`
         first
            second
               third
      `
    ).toMatchSnapshot();
  });

  describe("single line input", () => {
    it("works with single line input", () => {
      expect(dd`A single line of input.`).toMatchSnapshot();
    });

    it("works with single line and closing backtick on newline", () => {
      expect(dd`
        A single line of input.
      `).toMatchSnapshot();
    });

    it("works with single line and inline closing backtick", () => {
      expect(dd`
        A single line of input.`
      ).toMatchSnapshot();
    });
  });

  it("can be used as a function", () => {
    expect(dd(`
      A test argument.
    `)).toMatchSnapshot();
  });

  it("escapes backticks", () => {
    expect(dd`\``).toMatchSnapshot();
  });

  it("doesn't strip exlicit newlines", () => {
    expect(dd`
      <p>Hello world!</p>\n
    `).toMatchSnapshot();
  });

  it("doesn't strip exlicit newlines with mindent", () => {
    expect(dd`
      <p>
        Hello world!
      </p>\n
    `).toMatchSnapshot();
  });

  it("handles zero indent", () => {
    expect(dd`
First line
  Second line
    `).toMatchSnapshot();
  });

  it("ignores first line indent", () => {
    expect(dd`{
  int a;
  int b;
}`).toMatchSnapshot();
  });

  it("another nested text", () => {
    const decl = dd`{
  int a;
  int b;
}`;
    expect(dd`{
  int a;
  struct ${decl} foo;
  int b;
}`).toMatchSnapshot();
  });

});
