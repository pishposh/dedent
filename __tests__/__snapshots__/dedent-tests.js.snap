exports[`dedent another nested text 1`] = `
"{
  int a;
  struct {
    int a;
    int b;
  } foo;
  int b;
}"
`;

exports[`dedent can be used as a function 1`] = `"A test argument."`;

exports[`dedent doesn't strip exlicit newlines 1`] = `
"<p>Hello world!</p>
"
`;

exports[`dedent doesn't strip exlicit newlines with mindent 1`] = `
"<p>
  Hello world!
</p>
"
`;

exports[`dedent escapes backticks 1`] = `"\`"`;

exports[`dedent handles zero indent 1`] = `
"First line
  Second line"
`;

exports[`dedent ignores first line indent 1`] = `
"{
  int a;
  int b;
}"
`;

exports[`dedent single line input works with single line and closing backtick on newline 1`] = `"A single line of input."`;

exports[`dedent single line input works with single line and inline closing backtick 1`] = `"A single line of input."`;

exports[`dedent single line input works with single line input 1`] = `"A single line of input."`;

exports[`dedent works with blank first line 1`] = `
"Some text that I might want to indent:
  * reasons
  * fun
That\'s all."
`;

exports[`dedent works with dedent interpolation 1`] = `
"first
  second
  third
fourth"
`;

exports[`dedent works with inline interpolation 1`] = `
"first second
third fourth"
`;

exports[`dedent works with interpolation 1`] = `
"first line
second
third"
`;

exports[`dedent works with multiline interpolation 1`] = `
"first
  second
  third
fourth"
`;

exports[`dedent works with multiple blank first lines 1`] = `
"first
second
third"
`;

exports[`dedent works with removing same number of spaces 1`] = `
"first
   second
      third"
`;

exports[`dedent works with suppressed newlines 1`] = `
"first second
third"
`;

exports[`dedent works without interpolation 1`] = `
"first
second
third"
`;
